src/events/utils/calendarUtils.ts

```diff
@@ -1,105 +1,95 @@
-  const prev = date.minus({ months: 1 });
-  return getMonthLength(prev);
-}
-
-/**
- * Shift a Date object with a given number of months.
- * @param {Date} date JavaScript Date object to be shifted.
- * @param {number} amount The amount of which to shift the month of the Date object.
- * @returns {Date} Shifted Date object.
- */
-export function changeMonth(date: Date, amount: number) {
-  let dt = DateTime.fromJSDate(date);
-  if (amount > 0) {
-    dt = dt.plus({ months: amount });
-  } else if (amount < 0) {
-    dt = dt.minus({ months: Math.abs(amount) });
-  }
-  return dt.toJSDate();
-}
-
-/**
- * Formats a Date object to a nice formatting, e.g. 'April 2020'.
- * @param {Date} date JavaScript Date object.
- * @returns {string} Formated datestring.
- */
-export function getMonthAndYear(date: Date) {
-  const dt = DateTime.fromJSDate(date);
-  return dt.toFormat('MMMM yyyy');
-}
-
-/**
- * @summary Create the correct representation of the current month.
- * @description To easily display the events of the given month,
- * the month is represented as an Array of Arrays of Events.
- * The outer Array represents the ((day of the month) - 1) the event is on,
- * while the inner Array represents the events on that day.
- * @param {DateTime} month Current month.
- * @param {IEvent[]} events Events to inject into the month model.
- * @returns {IEvent[][]} Events represented in a month model.
- */
-export const constructMonthMap = (month: DateTime, events: IEvent[]): IEvent[][] => {
-  /**
-   * @summary Create an empty EventMonth.
-   * @description Create an array of length `daysInMonth`, containing empty arrays.
-   */
-  const map = [...Array(month.daysInMonth)].map(() => Array(0).fill([]));
-  events.forEach((event) => {
-    const day = DateTime.fromISO(event.start_date).day - 1;
-    map[day].push(event);
-  });
-  return map;
-};
+import { DateTime, Interval } from 'luxon';
+
+export function getFirstDayOfMonth(date: DateTime) {
+  return date.startOf('month');
+}
+
+export function getLastDayOfMonth(date: DateTime) {
+  return date.endOf('month');
+}
+
+export function getFirstDayOfWeek(date: DateTime): DateTime {
+  return date.startOf('week');
+}
+
+export function getLastDayOfWeek(date: DateTime): DateTime {
+  return date.endOf('week');
+}
+
+export function getStartOfDay(date: DateTime = DateTime.local()) {
+  return date.startOf('day');
+}
+
+/**
+ * Returns the weekday of the first day of a month from a date object.
+ * @param {DateTime} date A Luxon DateTime object.
+ * @returns {number} Zero indexed number between 0 and 6.
+ */
+export function getFirstWeekdayOfMonth(date: DateTime) {
+  /** Set the DateTime to the first day of the month */
+  const firstDay = getFirstDayOfMonth(date);
+  /** Get index of first day from DateTime, zero-index and return */
+  return firstDay.weekday - 1;
+}
+
+/**
+ * @param {DateTime} date Luxon DateTime object.
+ * @returns {number} Length of the given month. Number between 28 and 31.
+ */
```

Comment:
Although I'm all in favor of good documentation and comments, is this not a bit superfluous when we have defined the param- and return types through typescript?

Reply:
I agree, I didn't really add this now, I just hanged the file so much that it registered changes in all of it :sweat_smile:

Reply:
I ended up removing most of the functions now, since they were not in use